### PR TITLE
feat(view): add twig support

### LINKED ISSUE #707: Optionally support Twig

### PR DESCRIPTION
**Pull request by dinhquochan**
Hi,

This pull request introduces the integration of Twig as a templating engine into the Tempest framework. Key changes include adding Twig dependencies, creating configuration and initialization classes, and implementing a Twig-based view renderer.

Resolved #707 

BONUS: https://speakerdeck.com/fabpot/symfonycon-vienna-2025-twig-still-relevant-in-2025 :)

### CODE REVIEW

File: src/Tempest/View/src/Renderers/TwigConfig.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        public bool $debug = false,
+        public string $charset = 'utf-8',
+        public bool $strictVariables = false,
+        public string $autoescape = 'html',
+        public ?bool $autoReload = null,
+        public int $optimizations = -1,
+    ) {
+    }
+
+    public function toArray(): array
```
Comment (brendt) [inline src/Tempest/View/src/Renderers/TwigConfig.php:24]: I'd be fine with `(array) $this`, but this is good as well.

**Comment by brendt** [comment]
Awesome! The failing tests on windows happen because you need to use `assertStringEqualsStringIgnoringLineEndings` — could you change that still?

FTR: in the future, we plan to move blade and twig into separate packages, but that's outside the scope of this PR :)

Let me know when the line endings are fixed, happy to merge afterwards!

**Comment by dinhquochan** [comment]
> Awesome! The failing tests on windows happen because you need to use assertStringEqualsStringIgnoringLineEndings — could you change that still?

I switched to ￼`assertStringgEqualsStringIgnoringLineEndings`￼ and ￼`(array) $this)`￼ 👍 

> FTR: in the future, we plan to move blade and twig into separate packages, but that's outside the scope of this PR :)

Yes 🫡

**Comment by brendt** [comment]
Awesome! And thanks for writing the docs as well, really appreciate it!

**Pull request merged**